### PR TITLE
fix: switch to `@stylistic/exp-list-style` to resolve array edge-cases

### PR DESCRIPTION
**Pull request by susnux**
* closes #1166

### CODE REVIEW

**Comment by Antreesy** [comment]
IDK, are there options to keep/enforce multiline there, e.g. by line length?
image

**Comment by Antreesy** [comment]
This? https://eslint.style/rules/list-style#maxitems

Looks nice, I'd consider great if limited to 3 items, maybe?

**Comment by susnux** [comment]
> Looks nice, I'd consider great if limited to 3 items, maybe?

Pushed a commit

**Comment by ShGKme** [comment]
In the referenced PR the tests were adjusted to be more real life like, with som explanations

File: tests/fixtures/codestyle/output/array.js [hunk 1]
```diff
@@ -7,7 +7,13 @@ const items = [
...
 	'third',
 ]
-const LATIN_VOWELS = ['a', 'e', 'i', 'o', 'u']
+const LATIN_VOWELS = [
+	'a',
+	'e',
+	'i',
+	'o',
+	'u',
+]
```
[the remark this answers is not in the document]
Comment (susnux) [reply tests/fixtures/codestyle/output/array.js]: I adjusted it again and it will always use what the dev wanted to use.
It only enforces consistent newlines for opening tag (and closing).

Auto fix is always hard to get right but now you only have to set the first new line correctly to get it fixed like you want.

**Review by Antreesy — APPROVED** [review]
Like the result! Needs some adjustments in-app, but it keeps the formatting as you want it

File: tests/fixtures/codestyle/input/array.js [hunk 2]
```diff
@@ -1,16 +1,52 @@
...
+const HTML_FORM_ACTIONS = ['POST', 'GET']
+
+// ❌ Single-line array should have brackets on the same line while multi-line array should have brackets on the next line
+const USER_STATUSES = [
+	'online',
+	'away',
+	'dnd',
+	'invisible',
+	'offline']
+// 🚧 Currently this is an edge case and isn't fixed properly...
```
Comment (ShGKme) [inline tests/fixtures/codestyle/input/array.js]:
```suggestion

// ❌ Mixed single/multiline style is formatted based on the first element formatting
```

**Review by ShGKme — APPROVED** [review]
